README.md

# About OpenGUIlion

OpenGUIlion is a very basic application written in Python that only does one thing: it shows non-interactive, animated GUIs from the "Evangelion" franchise.

## Features

- Recreated interfaces from the computers and screens seen in "Evangelion", stretched to 2:1.
- Runs in Python 3.

## Requirements

To run OpenGUIlion you need Python 3 with PyGame and PIL available. Some distros already include them, but in case yours doesn't you can easily install them through `pip`.

You also need some extra fonts that must be placed under the `/openguilion/fonts/` folder and cannot be included in the repo for copyright reasons, whereas others can. How to get the copyrighted fonts is up to you and I will not provide further instructions nor direct downloads.

- Obtain "Helvetica Narrow Bold Regular" and name it `helvet.ttf`.
- Obtain "Helvetica Bold" and name it `helvetb.ttf`.
- Obtain "EVA-Matisse_Classic EB" and name it `matisse.ttf`. Since it has been published in the Internet Archive I do not consider illegal to link it. [Get it here.](https://archive.org/details/qjwi3h)

You could try to use Arial Narrow and Arial Bold as replacements for Helvetica, but I have not tested these cases.

## How to run

You can directly run `openguilion.py` from the command line. The program does not accept any arguments as of now. It has been tested under Python 3.9.

## Future developments

- Additional scenes.
- Adapt the code so the app can be properly displayed in Waveshare's 6in flexible AMOLED screen.

## About the author

This software has been made by [stage7](https://www.github.com/stage7). You can also [ping me on Twitter](https://twitter.com/s7age) if you need help or just say "hello".

The project's official Twitter account is [@OpenGUIlion](https://twitter.com/openguilion).

## License

This software is distributed under a [GPL 3.0 license](https://opensource.org/licenses/GPL-3.0).

## Disclaimer

All names are trademarks of their respective owners and are used here under fair use.

- "Evangelion" is a trademark of Gainax Corporation and Khara Inc.
- "Helvetica" is a trademark of Monotype Imaging Inc.
- "ITC Matisse®" is a trademark of Monotype ITC Inc.

Sinkin Sans is used under an [Apache License](https://www.fontsquirrel.com/license/sinkin-sans).


CHANGELOG.md

# OpenGUIlion change log

## Version 1.1 - 20220917
- [NEW] Scene: "state of emergency".
- [UPD] Some refactoring and new variables.

## Version 1.0 - 20220914

- [NEW] First public version, including five scenes: "case resolution", "lost", "intruder alert", "mental toxicity" and "phase 01 to 17".


openguilion.py
from openguilion.openguilionCommon import *
from openguilion.intruderAlert import *
from openguilion.caseResolution import *
from openguilion.phase01To17 import *
from openguilion.lost import *
from openguilion.mentalToxicity import *
from openguilion.stateEmergency import *

def main():
	running = True

	startDate = time.time()
	curTime = startDate
	animIndex = 0
	animTime = 0
	animations = [stateEmergency, intruderAlert, caseResolution, mentalToxicity, lost, phase01To17]

	while running:
		for event in pygame.event.get():
			keys = pygame.key.get_pressed()
			event = pygame.event.poll()
			if event.type == pygame.QUIT or keys[pygame.K_ESCAPE]:
				running = False
			if keys[pygame.K_a]:
				textX -= 1
				print(str(textX) + ", " + str(textY))
			if keys[pygame.K_d]:
				textX += 1
				print(str(textX) + ", " + str(textY))
			if keys[pygame.K_w]:
				textY -= 1
				print(str(textX) + ", " + str(textY))
			if keys[pygame.K_s]:
				textY += 1
				print(str(textX) + ", " + str(textY))

		animTime = time.time() - curTime
		if animations[animIndex % len(animations)](animTime):
			curTime = time.time()
			animIndex += 1
			animTime = 0

		pygame.display.update()

	pygame.quit()

if __name__ == "__main__":
	main()

openguilion/lost.py
from . import openguilionCommon as _
from PIL import Image, ImageDraw, ImageFont, ImageFilter, ImageChops
import pygame

# SURFACES
surfaceLost = pygame.Surface((_.w, _.h), pygame.SRCALPHA, 32)

# STATIC SCENE COMPOSITION
imgLost = _.fontHelvGiant.render("LOST", True, _.cBlack).convert_alpha()
imgLost = pygame.transform.smoothscale(imgLost, (_.w - 100, _.h + 80))

def lost(animTime):
	_.screen.fill(_.cBlack)
	surfaceLost.fill((0, 0, 0, 0))

	blinkTimeline = (animTime * 1000) % 750
	pygame.draw.rect(surfaceLost, _.cPureRed, (20, 20, _.w - 40, _.h - 40))
	if blinkTimeline <= 375:
		imgLost.set_alpha(255)
	else:
		imgLost.set_alpha(max(0, 255 - round((blinkTimeline - 300) / 375 * 255)))
	surfaceLost.blit(imgLost, (40, -10))

	pil_string_image = pygame.image.tostring(surfaceLost, "RGBA", False)
	pil_image = Image.frombytes("RGBA", (_.w, _.h), pil_string_image)
	draw = ImageDraw.Draw(pil_image)
	blurred = pil_image.filter(ImageFilter.BoxBlur(3))
	blurred = pygame.image.fromstring(blurred.tobytes(), blurred.size, blurred.mode)
	_.screen.blit(blurred, (0, 0, 0, 0))
	_.screen.blit(surfaceLost, (0, 0))

	if animTime >= 5:
		return True
	return False

openguilion/intruderAlert.py
from . import openguilionCommon as _
from PIL import Image, ImageDraw, ImageFont, ImageFilter, ImageChops
import pygame
import math
import random

# SURFACES
surfaceIntruderAlert = pygame.Surface((_.w, _.h), pygame.SRCALPHA, 32)
surfaceIntruderAlertBG = pygame.Surface((_.w, _.h), pygame.SRCALPHA, 32)
surfaceIntruderAlertOver = pygame.Surface((_.w, _.h), pygame.SRCALPHA, 32)

# STATIC SCENE COMPOSITION
imgUnidentInt = _.fontHelv.render("UNIDENTIFIED INTRUDER", True, _.cOrange).convert_alpha()
imgUnidentInt = pygame.transform.smoothscale(imgUnidentInt, (round(imgUnidentInt.get_size()[0] * 0.6), round(imgUnidentInt.get_size()[1] * 1)))
imgUnidentInt = pygame.transform.rotate(imgUnidentInt, 270)
imgSubcomp    = _.fontHelv.render("SUBCOMPUTER #1", True, _.cOrange).convert_alpha()
imgSubcomp    = pygame.transform.smoothscale(imgSubcomp, (round(imgSubcomp.get_size()[0] * 0.6), round(imgSubcomp.get_size()[1] * 0.65)))
imgIntAlert   = _.fontHelv.render("INTRUDER ALERT", True, _.cOrange).convert_alpha()
imgIntAlert   = pygame.transform.smoothscale(imgIntAlert, (round(imgIntAlert.get_size()[0] * 0.85), round(imgIntAlert.get_size()[1] * 1.6)))
imgPartition  = _.fontHelv.render("PARTITION", True, _.cOrange).convert_alpha()
imgPartition  = pygame.transform.smoothscale(imgPartition, (round(imgPartition.get_size()[0] * 0.3), round(imgPartition.get_size()[1] * 0.5)))
imgInt1       = _.fontHelv.render("#1", True, _.cOrange).convert_alpha()
imgInt1       = pygame.transform.smoothscale(imgInt1, (round(imgInt1.get_size()[0] * 0.5), round(imgInt1.get_size()[1] * 0.5)))
imgInt2       = _.fontHelv.render("#2", True, _.cOrange).convert_alpha()
imgInt2       = pygame.transform.smoothscale(imgInt2, (round(imgInt2.get_size()[0] * 0.5), round(imgInt2.get_size()[1] * 0.5)))
imgInt3       = _.fontHelv.render("#3", True, _.cOrange).convert_alpha()
imgInt3       = pygame.transform.smoothscale(imgInt3, (round(imgInt3.get_size()[0] * 0.5), round(imgInt3.get_size()[1] * 0.5)))
imgInt4       = _.fontHelv.render("#4", True, _.cOrange).convert_alpha()
imgInt4       = pygame.transform.smoothscale(imgInt4, (round(imgInt4.get_size()[0] * 0.5), round(imgInt4.get_size()[1] * 0.5)))
imgInt5       = _.fontHelv.render("#5", True, _.cOrange).convert_alpha()
imgInt5       = pygame.transform.smoothscale(imgInt5, (round(imgInt5.get_size()[0] * 0.5), round(imgInt5.get_size()[1] * 0.5)))
pygame.draw.rect(surfaceIntruderAlertBG, (186, 1, 13), (0, 0, 235, 512))
tempColor = pygame.Color(_.cBlack)
for i in range(235):
	tempColor.hsva = (172.74 + 127.26 * (i / 235), 100, 100, 50)
	pygame.draw.line(surfaceIntruderAlertBG, tempColor, (235 + i, 0), (235 + i, 511))
for i in range(554):
	tempColor.hsva = (300 - 300 * (i / 554), 100, 100, 50)
	pygame.draw.line(surfaceIntruderAlertBG, tempColor, (470 + i, 0), (470 + i, 511))
pygame.draw.rect(surfaceIntruderAlertBG, _.cBlack, (235 + 128, 0, 30, 512))
pygame.draw.rect(surfaceIntruderAlertBG, _.cBlack, (235 + 316, 0, 14, 512))
pygame.draw.rect(surfaceIntruderAlertBG, _.cBlack, (235 + 406, 0, 20, 512))
pygame.draw.rect(surfaceIntruderAlertBG, _.cBlack, (235 + 473, 0, 20, 512))
pygame.draw.rect(surfaceIntruderAlertBG, _.cBlack, (235 + 631, 0, 20, 512))
surfaceIntruderAlertBG.blit(imgUnidentInt, (142, 86))
surfaceIntruderAlertBG.blit(imgPartition, (265, 22))
surfaceIntruderAlertBG.blit(imgPartition, (435, 22))
surfaceIntruderAlertBG.blit(imgPartition, (600, 22))
surfaceIntruderAlertBG.blit(imgPartition, (762, 22))
surfaceIntruderAlertBG.blit(imgPartition, (913, 22))
surfaceIntruderAlertBG.blit(imgInt1, (292, 50))
surfaceIntruderAlertBG.blit(imgInt2, (461, 50))
surfaceIntruderAlertBG.blit(imgInt3, (626, 50))
surfaceIntruderAlertBG.blit(imgInt4, (787, 50))
surfaceIntruderAlertBG.blit(imgInt5, (939, 50))
surfaceIntruderAlertBG.blit(imgSubcomp, (496, 344))
surfaceIntruderAlertBG.blit(imgIntAlert, (499, 384))
pygame.draw.rect(surfaceIntruderAlertBG, _.cOrange, (144, 74, 61, 396), 3, border_radius = 6)
pygame.draw.rect(surfaceIntruderAlertBG, _.cOrange, (261, 20, 86, 60), 2, border_radius = 4)
pygame.draw.rect(surfaceIntruderAlertBG, _.cOrange, (431, 20, 86, 60), 2, border_radius = 4)
pygame.draw.rect(surfaceIntruderAlertBG, _.cOrange, (596, 20, 86, 60), 2, border_radius = 4)
pygame.draw.rect(surfaceIntruderAlertBG, _.cOrange, (758, 20, 86, 60), 2, border_radius = 4)
pygame.draw.rect(surfaceIntruderAlertBG, _.cOrange, (909, 20, 86, 60), 2, border_radius = 4)
pygame.draw.rect(surfaceIntruderAlertBG, _.cOrange, (489, 379, 391, 99), 5, border_radius = 8)

def intruderAlert(animTime):
	_.screen.fill(_.cBlack)
	surfaceIntruderAlert.fill((0, 0, 0, 0))
	surfaceIntruderAlertOver.fill((0, 0, 0, 0))

	surfaceIntruderAlert.blit(surfaceIntruderAlertBG, (0, 0))
	factor = animTime / 6
	if factor < 0.5:
		barsMultiplier = 16 * factor * factor * factor * factor * factor
	else:
		barsMultiplier = 1 - math.pow(-2 * factor + 2, 5) / 2
	
	offset = random.randint(-30, 30)
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (230, 256 - (120 + offset + 150 * barsMultiplier) / 2, 133, 120 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (362, 256 - (60 + offset + 150 * barsMultiplier) / 2, 22, 60 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (384, 256 - (72 + offset + 150 * barsMultiplier) / 2, 9, 72 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (393, 256 - (-18 + offset + 150 * barsMultiplier) / 2, 121, -18 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (514, 256 - (18 + offset + 150 * barsMultiplier) / 2, 30, 18 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (544, 256 - (30 + offset + 150 * barsMultiplier) / 2, 7, 30 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (551, 256 - (-48 + offset + 150 * barsMultiplier) / 2, 87, -48 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (638, 256 - (-30 + offset + 150 * barsMultiplier) / 2, 6, -30 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (644, 256 - (-12 + offset + 150 * barsMultiplier) / 2, 17, -12 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (661, 256 - (-90 + offset + 150 * barsMultiplier) / 2, 51, -90 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (712, 256 - (-60 + offset + 150 * barsMultiplier) / 2, 15, -60 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (727, 256 - (-90 + offset + 150 * barsMultiplier) / 2, 39, -90 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (766, 256 - (-96 + offset + 150 * barsMultiplier) / 2, 102, -96 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (868, 256 - (-78 + offset + 150 * barsMultiplier) / 2, 17, -78 + offset + 150 * barsMultiplier))
	pygame.draw.rect(surfaceIntruderAlertOver, _.cBrightRed, (885, 256 - (-114 + offset + 150 * barsMultiplier) / 2, 50, -114 + offset + 150 * barsMultiplier))
	surfaceIntruderAlertOver.set_alpha(64)
	pil_string_image = pygame.image.tostring(surfaceIntruderAlertOver, "RGBA", False)
	pil_image = Image.frombytes("RGBA", (_.w, _.h), pil_string_image)
	draw = ImageDraw.Draw(pil_image)
	blurred = pil_image.filter(ImageFilter.BoxBlur(10))
	blurred = pygame.image.fromstring(blurred.tobytes(), blurred.size, blurred.mode)
	blurred.set_alpha(192)
	surfaceIntruderAlert.blit(blurred, (0, 0, 0, 0))
	surfaceIntruderAlert.blit(surfaceIntruderAlertOver, (0, 0))

	_.screen.blit(surfaceIntruderAlert, (0, 0))

	if animTime >= 6:
		return True
	return False

openguilion/stateEmergency.py
from . import openguilionCommon as _
from PIL import Image, ImageDraw, ImageFont, ImageFilter, ImageChops
import pygame

# SURFACES
surfaceStateEmergency = pygame.Surface((_.w, _.h), pygame.SRCALPHA, 32)
surfaceStateEmergencyBG = pygame.Surface((_.w, _.h), pygame.SRCALPHA, 32)
surfaceStateEmergencyBGRects = pygame.Surface((_.w, _.h), pygame.SRCALPHA, 32)
surfaceStateEmergencyOver = pygame.Surface((_.w, _.h), pygame.SRCALPHA, 32)

# STATIC SCENE COMPOSITION
imgStateEm   = _.fontMatisse.render("非常事態", True, _.cRed).convert_alpha()
imgStateEm   = pygame.transform.smoothscale(imgStateEm, (round(imgStateEm.get_size()[0] * 2), round(imgStateEm.get_size()[1] * 1.5)))
imgEmergency = _.fontSinkin.render("EMERGENCY", True, _.cOrange).convert_alpha()
imgEmergency = pygame.transform.smoothscale(imgEmergency, (round(imgEmergency.get_size()[0] * 0.55), round(imgEmergency.get_size()[1] * 0.7)))

pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((43, 0), (113, 0), (0, 113), (0, 43)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((26, 159), (185, 0), (255, 0), (96, 159)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((168, 159), (327, 0), (397, 0), (238, 159)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((310, 159), (469, 0), (554, 0), (713, 159), (643, 159), (512, 28), (511, 28), (380, 159)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((452, 159), (511, 100), (512, 100), (571, 159)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((626, 0), (696, 0), (855, 159), (785, 159)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((768, 0), (838, 0), (997, 159), (927, 159)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((910, 0), (980, 0), (1023, 43), (1023, 113)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((0, 352), (40, 352), (0, 392)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((0, 464), (112, 352), (182, 352), (23, 511), (0, 511)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((254, 352), (324, 352), (165, 511), (95,511)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((396, 352), (466, 352), (307, 511), (237, 511)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((379, 511), (511, 379), (512, 379), (644, 511), (574, 511), (512, 449), (511, 449), (449, 511)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((557, 352), (627, 352), (786, 511), (716, 511)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((699, 352), (769, 352), (928, 511), (858, 511)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((841, 352), (911, 352), (1023, 464), (1023, 511), (1000, 511)))
pygame.draw.polygon(surfaceStateEmergencyBG, _.cRed, ((983, 352), (1023, 352), (1023, 392)))
pygame.draw.rect(surfaceStateEmergencyBG, _.cBlack, (512 - imgStateEm.get_size()[0] / 2 - 40, 35, imgStateEm.get_size()[0] + 80, imgStateEm.get_size()[1] + 20))
pygame.draw.rect(surfaceStateEmergencyBG, _.cBlack, (512 - imgStateEm.get_size()[0] / 2 - 40, _.h - 55 - imgStateEm.get_size()[1], imgStateEm.get_size()[0] + 80, imgStateEm.get_size()[1] + 20))

surfaceStateEmergencyBG.blit(imgStateEm, (512 - imgStateEm.get_size()[0] / 2, 45))
surfaceStateEmergencyBG.blit(imgStateEm, (512 - imgStateEm.get_size()[0] / 2, _.h - 45 - imgStateEm.get_size()[1]))


pygame.draw.rect(surfaceStateEmergencyBGRects, _.cOrange, (512 - imgStateEm.get_size()[0] / 2 - 35, 40, 20, imgStateEm.get_size()[1] + 10))
pygame.draw.rect(surfaceStateEmergencyBGRects, _.cOrange, (512 + imgStateEm.get_size()[0] / 2 + 15, 40, 20, imgStateEm.get_size()[1] + 10))
pygame.draw.rect(surfaceStateEmergencyBGRects, _.cOrange, (512 - imgStateEm.get_size()[0] / 2 - 35, _.h - 50 - imgStateEm.get_size()[1], 20, imgStateEm.get_size()[1] + 10))
pygame.draw.rect(surfaceStateEmergencyBGRects, _.cOrange, (512 + imgStateEm.get_size()[0] / 2 + 15, _.h - 50 - imgStateEm.get_size()[1], 20, imgStateEm.get_size()[1] + 10))

def stateEmergency(animTime):
	_.screen.fill(_.cBlack)
	surfaceStateEmergency.fill((0, 0, 0, 0))

	offsetEmergency = (-animTime * 500) % (imgEmergency.get_size()[0] + 350) - _.w
	while offsetEmergency < _.w:
		surfaceStateEmergency.blit(imgEmergency, (offsetEmergency, 180))
		offsetEmergency += (imgEmergency.get_size()[0] + 350)
	pygame.draw.rect(surfaceStateEmergency, _.cRed, (0, 321, _.w, 10))

	blinkTimeline = (animTime * 1000) % 500
	if blinkTimeline <= 250:
		surfaceStateEmergencyBG.set_alpha(255)
		surfaceStateEmergencyBGRects.set_alpha(255 - max(0, 255 - round((blinkTimeline - 200) / 250 * 255)))
	else:
		surfaceStateEmergencyBG.set_alpha(max(0, 255 - round((blinkTimeline - 200) / 250 * 255)))
		surfaceStateEmergencyBGRects.set_alpha(255)

	pil_string_image = pygame.image.tostring(surfaceStateEmergency, "RGBA", False)
	pil_image = Image.frombytes("RGBA", (_.w, _.h), pil_string_image)
	draw = ImageDraw.Draw(pil_image)
	blurred = pil_image.filter(ImageFilter.BoxBlur(6))
	blurred = pygame.image.fromstring(blurred.tobytes(), blurred.size, blurred.mode)
	_.screen.blit(blurred, (0, 0, 0, 0))
	surfaceStateEmergency.blit(surfaceStateEmergencyBG, (0, 0))
	surfaceStateEmergency.blit(surfaceStateEmergencyBGRects, (0, 0))
	surfaceStateEmergency.blit(surfaceStateEmergencyOver, (0, 0))

	_.screen.blit(surfaceStateEmergency, (0, 0))

	if animTime >= 5:
		return True
	return False

openguilion/phase01To17.py
from . import openguilionCommon as _
from PIL import Image, ImageDraw, ImageFont, ImageFilter, ImageChops
import pygame

# SURFACES
surfacePhase01to17 = pygame.Surface((_.w, _.h), pygame.SRCALPHA, 32)

# STATIC SCENE COMPOSITION
imgPhase01  = _.fontHelv.render("PHASE 01", True, _.cOrange).convert_alpha()
imgPhase02  = _.fontHelv.render("PHASE 02", True, _.cOrange).convert_alpha()
imgPhase03  = _.fontHelv.render("PHASE 03", True, _.cOrange).convert_alpha()
imgPhase04  = _.fontHelv.render("PHASE 04", True, _.cOrange).convert_alpha()
imgPhase05  = _.fontHelv.render("PHASE 05", True, _.cOrange).convert_alpha()
imgPhase06  = _.fontHelv.render("PHASE 06", True, _.cOrange).convert_alpha()
imgPhase07  = _.fontHelv.render("PHASE 07", True, _.cOrange).convert_alpha()
imgPhase08  = _.fontHelv.render("PHASE 08", True, _.cOrange).convert_alpha()
imgPhase09  = _.fontHelv.render("PHASE 09", True, _.cOrange).convert_alpha()
imgPhase10  = _.fontHelv.render("PHASE 10", True, _.cOrange).convert_alpha()
imgPhase11  = _.fontHelv.render("PHASE 11", True, _.cOrange).convert_alpha()
imgPhase12  = _.fontHelv.render("PHASE 12", True, _.cOrange).convert_alpha()
imgPhase13  = _.fontHelv.render("PHASE 13", True, _.cOrange).convert_alpha()
imgPhase14  = _.fontHelv.render("PHASE 14", True, _.cOrange).convert_alpha()
imgPhase15  = _.fontHelv.render("PHASE 15", True, _.cOrange).convert_alpha()
imgPhase16  = _.fontHelv.render("PHASE 16", True, _.cOrange).convert_alpha()
imgPhase17  = _.fontHelv.render("PHASE 17", True, _.cOrange).convert_alpha()
imgPhase01  = pygame.transform.smoothscale(imgPhase01, (round(imgPhase01.get_size()[0] * 0.3), round(imgPhase01.get_size()[1] * 0.4)))
imgPhase02  = pygame.transform.smoothscale(imgPhase02, (round(imgPhase02.get_size()[0] * 0.3), round(imgPhase02.get_size()[1] * 0.4)))
imgPhase03  = pygame.transform.smoothscale(imgPhase03, (round(imgPhase03.get_size()[0] * 0.3), round(imgPhase03.get_size()[1] * 0.4)))
imgPhase04  = pygame.transform.smoothscale(imgPhase04, (round(imgPhase04.get_size()[0] * 0.3), round(imgPhase04.get_size()[1] * 0.4)))
imgPhase05  = pygame.transform.smoothscale(imgPhase05, (round(imgPhase05.get_size()[0] * 0.3), round(imgPhase05.get_size()[1] * 0.4)))
imgPhase06  = pygame.transform.smoothscale(imgPhase06, (round(imgPhase06.get_size()[0] * 0.3), round(imgPhase06.get_size()[1] * 0.4)))
imgPhase07  = pygame.transform.smoothscale(imgPhase07, (round(imgPhase07.get_size()[0] * 0.3), round(imgPhase07.get_size()[1] * 0.4)))
imgPhase08  = pygame.transform.smoothscale(imgPhase08, (round(imgPhase08.get_size()[0] * 0.3), round(imgPhase08.get_size()[1] * 0.4)))
imgPhase09  = pygame.transform.smoothscale(imgPhase09, (round(imgPhase09.get_size()[0] * 0.3), round(imgPhase09.get_size()[1] * 0.4)))
imgPhase10  = pygame.transform.smoothscale(imgPhase10, (round(imgPhase10.get_size()[0] * 0.3), round(imgPhase10.get_size()[1] * 0.4)))
imgPhase11  = pygame.transform.smoothscale(imgPhase11, (round(imgPhase11.get_size()[0] * 0.3), round(imgPhase11.get_size()[1] * 0.4)))
imgPhase12  = pygame.transform.smoothscale(imgPhase12, (round(imgPhase12.get_size()[0] * 0.3), round(imgPhase12.get_size()[1] * 0.4)))
imgPhase13  = pygame.transform.smoothscale(imgPhase13, (round(imgPhase13.get_size()[0] * 0.3), round(imgPhase13.get_size()[1] * 0.4)))
imgPhase14  = pygame.transform.smoothscale(imgPhase14, (round(imgPhase14.get_size()[0] * 0.3), round(imgPhase14.get_size()[1] * 0.4)))
imgPhase15  = pygame.transform.smoothscale(imgPhase15, (round(imgPhase15.get_size()[0] * 0.3), round(imgPhase15.get_size()[1] * 0.4)))
imgPhase16  = pygame.transform.smoothscale(imgPhase16, (round(imgPhase16.get_size()[0] * 0.3), round(imgPhase16.get_size()[1] * 0.4)))
imgPhase17  = pygame.transform.smoothscale(imgPhase17, (round(imgPhase17.get_size()[0] * 0.3), round(imgPhase17.get_size()[1] * 0.4)))
imgPhases   = [imgPhase01, imgPhase02, imgPhase03, imgPhase04, imgPhase05, imgPhase06, imgPhase07, imgPhase08, imgPhase09, imgPhase10, imgPhase11, imgPhase12, imgPhase13, imgPhase14, imgPhase15, imgPhase16, imgPhase17]
imgStart    = _.fontHelv.render("START", True, _.cLightYellow).convert_alpha()
imgStart    = pygame.transform.smoothscale(imgStart, (round(imgStart.get_size()[0] * 0.8), round(imgStart.get_size()[1] * 0.5)))
imgDecontam = _.fontHelv.render("DECONTAM", True, _.cLightYellow).convert_alpha()
imgDecontam = pygame.transform.smoothscale(imgDecontam, (round(imgDecontam.get_size()[0] * 0.4), round(imgDecontam.get_size()[1] * 0.4)))
imgFinished = _.fontHelv.render("FINISHED", True, _.cLightYellow).convert_alpha()
imgFinished = pygame.transform.smoothscale(imgFinished, (round(imgFinished.get_size()[0] * 0.4), round(imgFinished.get_size()[1] * 0.4)))

def phase01To17(animTime):
	_.screen.fill((0, 0, 0))
	surfacePhase01to17.fill((0, 0, 0, 0))

	scrollY = round(animTime * 80)

	pygame.draw.rect(surfacePhase01to17, _.cOrange, (60, 0, 2, 512))
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (964, 0, 2, 512))
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (147, 55 - scrollY, 150, 32), border_radius = 4)
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (447, 55 - scrollY, 150, 32), border_radius = 4)
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (747, 55 - scrollY, 150, 32), border_radius = 4)
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (219, 0 - scrollY, 6, 1260))
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (519, 0, 6, 512))
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (819, 0 - scrollY, 6, 1260))
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (219, 1260 - scrollY, 606, 6))

	if animTime > 0.15:
		pygame.draw.rect(surfacePhase01to17, _.cLightYellow, (127, 57 - scrollY, 18, 32), border_radius = 2)
		pygame.draw.rect(surfacePhase01to17, _.cLightYellow, (427, 57 - scrollY, 18, 32), border_radius = 2)
		pygame.draw.rect(surfacePhase01to17, _.cLightYellow, (727, 57 - scrollY, 18, 32), border_radius = 2)
		pygame.draw.rect(surfacePhase01to17, _.cLightYellow, (301, 57 - scrollY, 18, 32), border_radius = 2)
		pygame.draw.rect(surfacePhase01to17, _.cLightYellow, (601, 57 - scrollY, 18, 32), border_radius = 2)
		pygame.draw.rect(surfacePhase01to17, _.cLightYellow, (901, 57 - scrollY, 18, 32), border_radius = 2)
		surfacePhase01to17.blit(imgStart, (157, 57 - scrollY))
		surfacePhase01to17.blit(imgStart, (457, 57 - scrollY))
		surfacePhase01to17.blit(imgStart, (757, 57 - scrollY))

	for i in range(17):
		surfacePhase01to17.blit(imgPhases[i], (67, 120 + i * 60 - scrollY))
		surfacePhase01to17.blit(imgPhases[i], (367, 120 + i * 60 - scrollY))
		surfacePhase01to17.blit(imgPhases[i], (667, 120 + i * 60 - scrollY))
		if ((animTime - 0.65) / 9) > (i / 17):
			pygame.draw.rect(surfacePhase01to17, _.cLightGreen, (147, 115 + i * 60 - scrollY, 150, 32), border_radius = 4)
			pygame.draw.rect(surfacePhase01to17, _.cLightGreen, (447, 115 + i * 60 - scrollY, 150, 32), border_radius = 4)
			pygame.draw.rect(surfacePhase01to17, _.cLightGreen, (747, 115 + i * 60 - scrollY, 150, 32), border_radius = 4)
		else:
			pygame.draw.rect(surfacePhase01to17, _.cRed, (147, 115 + i * 60 - scrollY, 150, 32), border_radius = 4)
			pygame.draw.rect(surfacePhase01to17, _.cRed, (447, 115 + i * 60 - scrollY, 150, 32), border_radius = 4)
			pygame.draw.rect(surfacePhase01to17, _.cRed, (747, 115 + i * 60 - scrollY, 150, 32), border_radius = 4)

	pygame.draw.rect(surfacePhase01to17, _.cOrange, (147, 115 + 17 * 60 - scrollY, 150, 64), border_radius = 4)
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (447, 115 + 17 * 60 - scrollY, 150, 64), border_radius = 4)
	pygame.draw.rect(surfacePhase01to17, _.cOrange, (747, 115 + 17 * 60 - scrollY, 150, 64), border_radius = 4)
	if ((animTime - 0.65) / 9) > 1:
		surfacePhase01to17.blit(imgDecontam, (165, 124 + 17 * 60 - scrollY))
		surfacePhase01to17.blit(imgFinished, (174, 148 + 17 * 60 - scrollY))
		surfacePhase01to17.blit(imgDecontam, (465, 124 + 17 * 60 - scrollY))
		surfacePhase01to17.blit(imgFinished, (474, 148 + 17 * 60 - scrollY))
		surfacePhase01to17.blit(imgDecontam, (765, 124 + 17 * 60 - scrollY))
		surfacePhase01to17.blit(imgFinished, (774, 148 + 17 * 60 - scrollY))

	pil_string_image = pygame.image.tostring(surfacePhase01to17, "RGBA", False)
	pil_image = Image.frombytes("RGBA", (_.w, _.h), pil_string_image)
	draw = ImageDraw.Draw(pil_image)
	blurred = pil_image.filter(ImageFilter.BoxBlur(3))
	blurred = pygame.image.fromstring(blurred.tobytes(), blurred.size, blurred.mode)
	_.screen.blit(blurred, (0, 0, 0, 0))
	_.screen.blit(surfacePhase01to17, (0, 0))

	if animTime >= 10:
		return True
	return False